cfep-22.md

```diff
@@ -0,0 +1,183 @@
+patch level (`x.x.x`) in order to prevent ABI breaks.
+
+Recipes whose package version is the same as the ABI version do not need to
+modify their build string.
+
+Recipes with multiple library artifacts whose ABIs are tracked separately
+should split these libraries into separate outputs of one recipe. A metapackage
+may be used for installation convenience, but the metapackge must also
+enumerate the run_exports for each subpackage.
+
+## Sample Implementation
+
+```yaml
+{% set name = "libavif" %}
+{% set build = 0 %}
+# NOTE: Humans must also update the library version in the tests section of this recipe
+{% set version = "0.10.1" %}
+# Look in the libavif top level CMakeLists.txt for the updated ABI version.
+# ABI is updated separately from API version.
+{% set so_version = "14.0.1" %}
+{% set so_major_version = so_version.split('.')[0] %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+build:
+  number: {{ build }}
+  string: "v{{ so_name_major }}soh{{ PKG_HASH }}_{{ build }}"
+  run_exports:
+    - {{ pin_subpackage(name|lower) }} *v{{ so_name_major }}so*
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libavif.so.{{ so_major_version }}     # [linux]
+    - test -f ${PREFIX}/lib/libavif.so.{{ so_version }}           # [linux]
+
+```
+
+## Rationale
+
+Adding the ABI version to the build string allows the run_export to handle both
+API and ABI using existing conda capabilities. Not needing new conda features
+saves engineering effort and is implementable today. Build strings are already
+used to track features such as the CUDA version and MPI variants.
+
+The ABI version is sandwiched between `v` and `so` because these letters are
+not part of hexadecimal and to prevent matching collisions with other build
+string tags or higher numbers. i.e. `6*` matches with `60`, but `6so*` does
+not.
+
+Using a combination of build string and run_exports to add ABI information
+requires no changes from downstream packages except using the latest build.
+They will be guarded from ABI breakages as soon as they rebuild with the
+updated package.
+
+## Backward Compatability
+
+This proposal does not break exisiting packages. It is a feature that only
+helps future package builds.
+
+
+## Alternatives
+
+### API Pinning Only
+
+In theory, breaking ABI changes can be introduced without changing the API
+(reordering struct elements for example). In practice, project managers would
+always make a new API release for this change, so ABI breaks can be avoided by
+pinning down to the patch version. This is the most conservative approach, but
+is the least flexible. Recipe maintainers can pin to minor API versions if the
+upstream package makes any such promises about not breaking the ABI.
+
+### Adding ABI name to package name
+
+This alternatives renames outputs to `{{ name|lower }}{{ so_major_version }}`.
+This is not backward compatible and would require migrating all downstream
+packages to the new output names.
```

Comment:
```suggestion
### Adding ABI name to package name

This alternatives renames outputs to `{{ name|lower }}{{ so_major_version }}`.
This is not backward compatible and would require migrating all downstream
packages to the new output names. However, it does enable installing multiple ABI versions simultaneously (if you don't mind refactoring to prevent clobbering).
```

> Why not go with the prior art, which is to have a libgfortran5 output? This is what every other binary distro does.

I have addressed this in the Alternatives section. It is also discussed in one of the linked issues: https://github.com/conda-forge/conda-forge.github.io/issues/157

An upside to this approach which I have added is that you can install multiple versions simultaneously.